### PR TITLE
RELATED: ONE-4163 Add locale option to DateFilter component

### PR DESCRIPTION
**Pull request by petrjanu**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5152/ : SUCCESS in 4m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/581/ : SUCCESS in 6s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5153/ : SUCCESS in 5m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/582/ : SUCCESS in 6s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2692/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/583/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2127/ : UNSTABLE in 12m 48s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5154/ : SUCCESS in 7m 42s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/584/ : SUCCESS in 6s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2693/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/585/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2128/ : SUCCESS in 12m 33s

File: src/components/filters/DateFilter/DateFilter.tsx [hunk 1]
```diff
@@ -30,12 +30,13 @@ interface IStatePropsIntersection {
...
 }
 
-export interface IDateFilterStateProps extends IStatePropsIntersection {
+export interface IDateFilterOwnProps extends IStatePropsIntersection {
     filterOptions: ExtendedDateFilters.IDateFilterOptionsByType;
     availableGranularities: ExtendedDateFilters.DateFilterGranularity[];
     isEditMode?: boolean;
     customFilterName?: string;
     dateFilterMode: ExtendedDateFilters.DashboardDateFilterConfigMode;
+    locale?: string;
```
Comment (vlastikcz) [inline src/components/filters/DateFilter/DateFilter.tsx]: what will happen for invalid locales?

File: stories/internal/DateFilter/DateFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+// (C) 2007-2019 GoodData Corporation
```
Comment (ivan-nejezchleb) [inline stories/internal/DateFilter/DateFilter.tsx]: why this file was created? There is already some DateFilter storybook in Helper components section, same as Attribute Filter. Why don't put it there?

File: stories/internal/DateFilter/DateFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                <DateFilterController />
+            </div>,
+        );
+    })
+    .add("localized", () => {
+        class DateFilterController extends React.Component<{}, {}> {
+            public render() {
+                return (
+                    <DateFilter
+                        locale={"de-DE"}
```
Comment (ivan-nejezchleb) [inline stories/internal/DateFilter/DateFilter.tsx]: unnecessary {}

File: stories/internal/DateFilter/DateFilter.tsx (hunk 2 above)
Comment (petrjanu) [reply stories/internal/DateFilter/DateFilter.tsx]: Git shows this as new file, but actually it is renamed file `DateFilterCore.tsx` to this new name `DateFilter.tsx`. So there was one story, I added second story named "localized".

File: src/components/filters/DateFilter/DateFilter.tsx (hunk 1 above)
Comment (petrjanu) [reply src/components/filters/DateFilter/DateFilter.tsx]: You will got console error, plus there will be localization IDs used instead of translation.

Comment (petrjanu) [reply src/components/filters/DateFilter/DateFilter.tsx]: But there is note in documentation `The localization of the component. Defaults to `en-US`. For other languages, see the [full list of available localizations](https://github.com/gooddata/gooddata-react-components/tree/master/src/translations).`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5160/ : SUCCESS in 5m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/597/ : SUCCESS in 16s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2699/ : SUCCESS in 16m 52s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5162/ : SUCCESS in 4m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1265/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1714/ : SUCCESS in 17m 40s

**Pull request merged**